sdk/resourcemanager/resources/armresources/cae_mock_test.go

```diff
@@ -0,0 +1,72 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package armresources_test
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/mock"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/test/credential"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+	"github.com/stretchr/testify/require"
+)
+
+const caeChallenge = `Bearer realm="", error_description="Continuous access evaluation resulted in challenge", error="insufficient_claims", claims="eyJhY2Nlc3NfdG9rZW4iOnsibmJmIjp7ImVzc2VudGlhbCI6dHJ1ZSwgInZhbHVlIjoiMTcyNjI1ODEyMiJ9fX0=" `
```

Comment:
will this be a security problems to include the value of 'claims' in the code?

Reply:
this solution is from mingzhe's suggestion. .net has similar test and it seems there is no security detection reported: https://github.com/Azure/azure-sdk-for-net/pull/47138

Reply:
also, similar tests in go azcore: https://github.com/Azure/azure-sdk-for-go/blob/77e8721c8a2b7e8be0eb5c01945e10cf5ed68764/sdk/azcore/runtime/policy_bearer_token_test.go#L384